Tests/FoundationTests/DateExtensionsTests.swift

```diff
@@ -970,6 +970,31 @@ final class DateExtensionsTests: XCTestCase {
         let tomorrowCheck = Calendar.current.date(byAdding: .day, value: 1, to: date)
         XCTAssertEqual(tomorrow, tomorrowCheck)
     }
+
+    func testDayInYear() {
+        let date = Date()
+        let dayInYear = date.dayInYear
+        let dayInYearCheck = Calendar.current.ordinality(of: .day, in: .year, for: date)!
+        XCTAssertEqual(dayInYear, dayInYearCheck)
+    }
+
+    func testNumberOfDaysInYear() {
+        let date = Date()
+        let numberOfDaysInYear = date.numberOfDaysInYear
+        let numberOfDaysInYearCheck = (Calendar.current.range(of: .day, in: .year, for: date)!.upperBound) - 1
+        XCTAssertEqual(numberOfDaysInYear, numberOfDaysInYearCheck)
+    }
+
+    func testnNmberOfDaysLeftInYear() {
+        let date = Date()
+        let dayLeftInAYear = date.numberOfDaysLeftInYear
+
+        let dayInYear = Calendar.current.ordinality(of: .day, in: .year, for: date)!
+        let numberOfDaysInYear = (Calendar.current.range(of: .day, in: .year, for: date)!.upperBound) - 1
+        let dayLeftInAYearCheck = numberOfDaysInYear - dayInYear
+
+        XCTAssertEqual(dayLeftInAYear, dayLeftInAYearCheck)
+    }
```

Comment:
This is not a valid way to do unit tests. You're performing the function and what you know is its implementation and checking that they're equal, which logically they will be (unless someone changes either the test or the implementation).
What you should be doing instead is using fixed values, i.e. a specific date where you know how many days are in that year, which day of the year it is and how many days left in that year.

Reply:
+1 Here

Reply:
@guykogus Many thanks for the comment!
I'll re-write the tests to make it correct :)